### PR TITLE
simBundle & recent naming changes

### PR DESCRIPTION
**Pull request by zeroXbrock**
- update event scheme; add mevGasPrice & gasUsed
- add SimBundleOptions, add builders to sendTransaction params
- fix munger bugs, add mungeSimBundle
- fix broken example
- update matchmaker api: simulateBundle
- remove unnecessary logs, unnecessary sim option from example
- define types for & munge sendBundle, simBundle responses

### CODE REVIEW

File: src/examples/sendBackrunBundle.ts [hunk 1]
```diff
@@ -43,41 +44,61 @@ const handleBackrun = async (
     provider: JsonRpcProvider,
     matchmaker: Matchmaker,
     pendingMutex: Mutex,
-) => {
-    console.log("pending tx", pendingTx)
-    const targetBlock = await provider.getBlockNumber() + 2
+    pendingTxHashes: AsyncArray<string>,
+): Promise<void> => {
+    pendingTx.mevGasPrice
+    if (!pendingTxHashes.includes(pendingTx.hash)) {
```
Comment (epheph) [inline src/examples/sendBackrunBundle.ts:50]: This is a promise, needs await?

File: src/matchmaker.ts [hunk 2]
```diff
@@ -149,7 +170,7 @@ export default class Matchmaker {
      * @param params Parameters for the bundle.
      * @returns Simulation data object.
      */
-    public async simulateBundle(params: BundleParams): Promise<any> {
+    public async simulateBundle(params: BundleParams, simOptions?: SimBundleOptions): Promise<ISimBundleResult> {
```
Comment (epheph) [inline src/matchmaker.ts:173]: the jsdoc doesn't match the func now

**Review by epheph — APPROVED** [review]
left 2 comments, but after fixing, lgtm.

File: src/matchmaker.ts (hunk 2 above)
Comment (zeroXbrock) [reply src/matchmaker.ts:174]: good catch, fixed

File: src/examples/sendBackrunBundle.ts (hunk 1 above)
Comment (zeroXbrock) [reply src/examples/sendBackrunBundle.ts:50]: oh I see, I was looking at the following line. thanks, fixed

**Pull request merged**